lib/Dialect/FIRRTL/Transforms/LegacyWiring.cpp

```diff
@@ -65,14 +65,10 @@ LogicalResult circt::firrtl::applyWiring(const AnnoPathValue &target,
     }
   } else if (auto opResult = target.ref.dyn_cast<OpAnnoTarget>()) {
     if (target.isOpOfType<WireOp, RegOp, RegResetOp>()) {
-      auto module = cast<FModuleOp>(opResult.getModule());
-      if (clazz == wiringSourceAnnoClass) {
-        builder.setInsertionPointToStart(module.getBodyBlock());
-      } else if (clazz == wiringSinkAnnoClass) {
-        builder.setInsertionPointToEnd(module.getBodyBlock());
-      }
-      targetValue = getValueByFieldID(builder, opResult.getOp()->getResult(0),
-                                      target.fieldIdx);
+      auto *targetBase = opResult.getOp();
+      builder.setInsertionPointAfter(targetBase);
```

Comment:
Possibly:

```c++
builder.setInsertionPointAfterValue(opResult);
```

Reply:
Whoa!! That exists?!?

```
    /// Sets the insertion point to the node after the specified value. If value                                                                                                                                                                                            
    /// has a defining operation, sets the insertion point to the node after such                                                                                                                                                                                           
    /// defining operation. This will cause subsequent insertions to go right                                                                                                                                                                                               
    /// after it. Otherwise, value is a BlockArgument. Sets the insertion point to                                                                                                                                                                                          
    /// the start of its block.
    void setInsertionPointAfterValue(Value val) {
```

Nice!!

Reply:
Unfortunately, `opResult` is not an `OpResult` despite the name, which would work!  (it's an `OpAnnoTarget`).

Still, thanks for the suggestion, TIL!